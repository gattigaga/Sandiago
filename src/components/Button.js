import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { TouchableNativeFeedback } from "react-native";

const Container = styled.View`
  width: 100%;
  padding-vertical: 16px;
  background-color: ${({ isInverted }) => (isInverted ? "white" : "#474787")};
`;

const Text = styled.Text`
  font-family: Roboto-Bold;
  font-size: 18px;
  text-align: center;
  color: ${({ isInverted }) => (isInverted ? "#474787" : "white")};
`;

const Button = ({ style, isInverted, caption, onPress }) => (
  <TouchableNativeFeedback onPress={onPress}>
    <Container style={style} isInverted={isInverted}>
      <Text isInverted={isInverted}>{caption}</Text>
    </Container>
  </TouchableNativeFeedback>
);

Button.propTypes = {
  style: PropTypes.array,
  isInverted: PropTypes.bool,
  caption: PropTypes.string,
  onPress: PropTypes.func
};

Button.defaultProps = {
  caption: "Button"
};

export default Button;
