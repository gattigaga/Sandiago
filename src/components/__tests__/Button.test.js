import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Button from "../Button";

describe("Button", () => {
  const setup = propOverrides => {
    const props = {
      caption: "Login",
      onPress: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<Button {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders with custom style", () => {
    const { wrapper } = setup({ style: [{ backgroundColor: "red" }] });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should calls 'onPress' while pressed", () => {
    const { wrapper, props } = setup();

    wrapper.simulate("press");

    expect(props.onPress).toBeCalled();
  });
});
