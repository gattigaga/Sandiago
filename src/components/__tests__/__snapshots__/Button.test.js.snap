// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should renders default 1`] = `
<DummyTouchableNativeFeedback
  onPress={[MockFunction]}
>
  <Styled(Component)>
    <Styled(Text)>
      Login
    </Styled(Text)>
  </Styled(Component)>
</DummyTouchableNativeFeedback>
`;

exports[`Button should renders with custom style 1`] = `
<DummyTouchableNativeFeedback
  onPress={[MockFunction]}
>
  <Styled(Component)
    style={
      Array [
        Object {
          "backgroundColor": "red",
        },
      ]
    }
  >
    <Styled(Text)>
      Login
    </Styled(Text)>
  </Styled(Component)>
</DummyTouchableNativeFeedback>
`;
